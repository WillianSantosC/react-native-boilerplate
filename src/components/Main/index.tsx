import { Text, View } from 'react-native';
const Main = () => {
  return (
    <View className="flex-1 items-center justify-center">
      <Text accessibilityRole="header" className="text-xl font-bold text-blue-600">
        Hello World
      </Text>
    </View>
  );
};

export default Main;
