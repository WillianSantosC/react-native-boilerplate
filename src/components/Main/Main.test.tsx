import { render } from '@testing-library/react-native';
import Main from '.';

describe('Main', () => {
  it('should render correctly', () => {
    const { getByText } = render(<Main />);
    const textElement = getByText('Hello World');
    expect(textElement).toBeTruthy();
  });

  it('should have accessibilityRole header', () => {
    const { getByAccessibilityHint } = render(<Main />);
    const header = getByAccessibilityHint('header');
    expect(header).toBeTruthy();
    expect(header.props.children).toBe('Hello World');
  });

  it('should match snapshot', () => {
    const tree = render(<Main />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
