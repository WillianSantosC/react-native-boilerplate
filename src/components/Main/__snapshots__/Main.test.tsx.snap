// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match snapshot 1`] = `
<View
  className="flex-1 items-center justify-center"
>
  <Text
    accessibilityHint="header"
    className="text-xl font-bold text-blue-600"
  >
    Hello World
  </Text>
</View>
`;
